source/server/active_tcp_socket.cc

```diff
@@ -89,7 +111,14 @@ void ActiveTcpSocket::continueFilterChain(bool success) {
           no_error = false;
           break;
         } else {
-          // Blocking at the filter but no error
+          // When reach this branch, the maxReadBytes must greater than 0.
+          // the negative maxReadBytes is invalid.
+          ASSERT((*iter_)->maxReadBytes() > 0);
+          // There may already have data peeked due to previous filter.
+          if (listener_filter_buffer_ != nullptr) {
+            listener_filter_buffer_->activateFileEvent(Event::FileReadyType::Read);
```

Comment:
Can you describe more what this case handles? How would there be a previous filter, but we haven't activated the read event?

Reply:
Yea, I will update